libs/@local/hql/frontend-jexpr/src/parser/array.rs

```diff
@@ -0,0 +1,82 @@
+use std::assert_matches::debug_assert_matches;
+
+use hql_diagnostics::Diagnostic;
+use hql_span::{SpanId, TextRange};
+
+use super::stream::TokenStream;
+use crate::{
+    lexer::{syntax_kind::SyntaxKind, token::Token, token_kind::TokenKind},
+    parser::error::unexpected_token,
+    span::Span,
+};
+
+/// Parse an array from the lexer
+///
+/// Assumes that the lexer has already consumed the opening bracket.
+///
+/// # Panics
+///
+/// Panics if the lexer has not consumed the opening bracket.
+// TODO: test `[]`, `[1]`, `[1, 2]`, error on `[1,]`, `[1, 2,]`, `[,1]`, `[,]`
+// TODO: quickcheck
```

Comment:
```suggestion
```

Reply:
removed todos in [4860af0](https://github.com/hashintel/hash/pull/4828/commits/4860af01f142c40a32fdcd87ac6e373133dd8366)